plugins/voxeldownsize/filter/VoxelDownsizeFilter.cpp

```diff
@@ -108,10 +110,10 @@ PointViewSet VoxelDownsizeFilter::run(PointViewPtr view)
 
 bool VoxelDownsizeFilter::find(int gx, int gy, int gz)
 {
-    //m_pool->await();
     auto itr = m_populatedVoxels.find(std::make_tuple(gx, gy, gz));
     if (itr == m_populatedVoxels.end())
     {
+        m_pool->await();
```

Comment:
I think I would prefer to use a lock rather than an await statement. It seems a bit more generic and safer, what do you think?

Reply:
await(), actually does the same.
It holds the lock, until all the threads from the pool are completed.
I think we don't need to implement that separately.